pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -132,20 +133,26 @@ func (d *DependenciesDistributor) OnAdd(obj interface{}) {
 
 // OnUpdate handles object update event and push the object to queue.
 func (d *DependenciesDistributor) OnUpdate(oldObj, newObj interface{}) {
-	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	unstructuredNewObj, err := helper.ToUnstructured(newObj)
 	if err != nil {
-		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+	if !unstructuredNewObj.GetDeletionTimestamp().IsZero() {
```

Comment:
I remember, if a resource doesn't have `Finalizers`, then when it is deleted, there won't be any update events.

Reply:
what this PR wants to do is to ignore the resources that need to be deleted.